### PR TITLE
Determine the bot name using the GitHub API

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Instead of hard-coding the "k8s-merge-robot" name into the mungers, we
need to read the name of the authenticated user from GitHub once we have
configured the client.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

Fixes https://github.com/kubernetes/test-infra/issues/3677

/area mungegithub

/cc @kargakis @eparis

### CODE REVIEW

File: mungegithub/github/github.go [hunk 1]
```diff
@@ -471,6 +475,20 @@ func (config *Config) PreExecute() error {
...
 	}
 	config.ResetAPICount()
+
+	// passing an empty username returns information
+	// about the currently authenticated user
+	username := ""
+	user, _, err := config.client.Users.Get(context.Background(), username)
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve currently authenticatd user: %v", err)
+	} else if user == nil {
```
Comment (stevekuznetsov) [inline mungegithub/github/github.go]: Not sure if we can get `nil` for `User` or `User.Login` if we get  no error on the `Get()`, so not clear if these are necessary

Comment (stevekuznetsov) [inline mungegithub/github/github.go]: OK! let's keep these in, then.

**Comment by eparis** [comment]
given it is github.go that owns the connection to github, should the 'user' be a function of github.go (aka the 'config' instead of storing the user in all the mungers?

**Comment by stevekuznetsov** [comment]
> given it is github.go that owns the connection to github, should the 'user' be a function of github.go (aka the 'config' instead of storing the user in all the mungers?

@eparis could you point me to a place in the current code where a munger gets info from `github.go` like that? Currently it _is_ a field of the `github.Config` but the way I saw to plumb it through on first blush was to copy it over in `Initialize(config github.Config...)`

**Comment by eparis** [comment]
ok, i see what you are saying. You usually have the mungeobject, not a config.  We could (haven't decided if I like it) have an accessor method for `obj.config.botName`   obj.BotName().  but it feels like a violation. That's not what the obj is supposed to be for...   Maybe I'd feel better if you just stored the config in the munger instead of just the name.

Even cleaner could be `obj.IsOurComment(comment)` and let github.go get the botname out of the `config`

Or more generic `obj.IsUs(*github.User)` ?

**Comment by kargakis** [comment]
@eparis the munger Object already contains a Config

**Comment by stevekuznetsov** [comment]
Definitely cleaner. @eparis PTAL

**Comment by stevekuznetsov** [comment]
Fixes https://github.com/kubernetes/test-infra/pull/3739

File: boskos/ranch/ranch_test.go [hunk 2]
```diff
@@ -228,7 +228,7 @@ func TestAcquireRoundRobin(t *testing.T) {
 		}
 	}
 
-	for idx, _ := range c.Resources {
```
Comment (stevekuznetsov) [inline boskos/ranch/ranch_test.go]: Does this repo not use `gofmt -s` ? I can undo these accidental changes if so.

**Comment by stevekuznetsov** [comment]
@eparis you've given LGTM and tests are green -- what is the next step here?

**Comment by kargakis** [comment]
@stevekuznetsov needs a rebase

File: boskos/ranch/ranch_test.go (hunk 2 above)
Comment (spxtr) [inline boskos/ranch/ranch_test.go]: [Doesn't look like it](https://github.com/kubernetes/test-infra/blob/b59e249f7821090ade1d0949dc09a5d18fe17423/verify/verify-gofmt.sh), but feel free to keep the changes.

Comment (kargakis) [inline boskos/ranch/ranch_test.go]: Can we also update that script with `-s` to keep these changes consistent with future changes?

**Comment by stevekuznetsov** [comment]
@kargakis done

**Comment by kargakis** [comment]
@stevekuznetsov can you rebase on HEAD since the gofmt changes are already merged?

**Comment by stevekuznetsov** [comment]
It will drop out in the merge, but sure

**Comment by kargakis** [comment]
Thanks, LGTM too

Fixes https://github.com/kubernetes/test-infra/issues/3677

**Pull request merged**